os/fs/smartfs/smartfs_utils.c

```diff
@@ -1255,10 +1255,15 @@ int smartfs_find_availableentry(struct smartfs_mountpt_s *fs, uint16_t parentdir
 	}
 
 	if (ret != OK) {
-		return -ENOENT;
+		/* No available entry was found, so we create one */
+		*new_chain = TRUE;
+		ret = smartfs_createentry(fs, direntry->dsector, direntry);
```

Comment:
hmm... I checked PR #4406 I think we'd better keep 3 parameter. (2nd one is for parentsector)

Reply:
direntry->dsector is the last  sector in the parent chain that was scanned for available entries.
Since no  entries were found, we know we have to add a sector at the END of the chain.

Direntry->dsector points to the last sector in the chain,
So we directly know where to chain new sector.
We don't have to traverse entire parent chain to get to last sector for chaining.